### DISCUSSION TITLE
How to terminate `par-each`

### DISCUSSION THREAD

**Discussion by cmpute** [nushell/nushell#7152] [Q&A] [answered]
How do I terminate all the commands running with `par-each`? When I press ctrl+c, it seems that nushell only sends the signal to one of the tasks, rather than all of them.

**Comment by rgwood** [comment, marked as answer]
Hi, do you have an example of a command that can't be terminated this way? What OS and Nushell version are you on?

If I run `[3sec 3sec 3sec] | par-each { |it| sleep $it; $"slept ($it)s" }` it terminates as expected with a single `ctrl+c`.

**Reply by cmpute** [reply]
You can try: `..3 | par-each { python -c "import time, itertools; [(print(i), time.sleep(0.1)) for i in itertools.count()]" }`

My OS is the latest KDE Neon and nushell version is 0.71

**Reply by rgwood** [reply]
Thanks. I am able to reproduce this on Linux but not Windows. I'll convert this to an issue.

**Reply by rgwood** [reply]
https://github.com/nushell/nushell/issues/7154
